jax/__init__.py

```diff
@@ -149,6 +149,7 @@
 
 # These submodules are separate because they are in an import cycle with
 # jax and rely on the names imported above.
+from jax import custom_call as custom_call
```

Comment:
Should we, perhaps, expose this somewhere under `jax.extend` or even not expose at all until we are sure this is useful? 

CC @froystig

Reply:
I think either of these options would make sense. I don't have strong opinions either way!

Reply:
I've moved this interface to `jax.extend.custom_call`.